# Problemas de prueba de hipótesis

## Ejercicio 1

Usted es un nutricionista que investiga dos tipos diferentes de dietas para ver si existe una diferencia significativa en la pérdida de peso después de un mes. Eliges dos grupos aleatorios de personas; un grupo sigue la primera dieta y el otro sigue la segunda. Al final del mes, se registra la pérdida de peso (en kg) de cada persona:

| Diet 1 | Diet 2 |
|:-------|:-------|
| 2.0 | 3.0 |
| 2.5 | 3.2 |
| 3.0 | 3.1 |
| 2.8 | 2.9 |
| 2.3 | 2.8 |
| 2.7 | 3.0 |
| 2.5 | 3.2 |

Con estos datos se busca responder a la siguiente pregunta: ¿Existe una diferencia significativa en la pérdida de peso promedio entre las personas que siguieron la primera dieta y las que siguieron la segunda dieta?

Para sacar conclusiones, siga los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba para comprobar la hipótesis. Puede utilizar la prueba t de Student.
-Analizar las conclusiones.

In [40]:
import scipy.stats as stats

#H0 = No hay diferencia significativa en la pérdida de peso promedio entre ambas dietas
#H1 = Existe una diferencia significativa en la pérdida de peso promedio entre ambas dietas

# Datos
dieta_1 = [2.0, 2.5, 3.0, 2.8, 2.3, 2.7, 2.5]
dieta_2 = [3.0, 3.2, 3.1, 2.9, 2.8, 3.0, 3.2]

# Prueba t para dos muestras independientes
t_stat, p_value = stats.ttest_ind(dieta_1, dieta_2,equal_var=False)

print(f"t-value: {t_value:.3f}")
print(f"p-value: {p_value:.3f}")


t-value: -3.538
p-value: 0.007


Obtenemos los siguientes resultados:

Estadístico t = -3.54

p-valor = 0.0071

Tomamos una significancia común de 
𝛼=0.05

Dado que el p-valor (0.0071) es menor que 0.05, rechazamos la hipótesis nula 𝐻0.

Conclusión:
Existe una diferencia estadísticamente significativa en la pérdida de peso promedio entre las personas que siguieron la Dieta 1 y la Dieta 2. Es decir, las dietas no tienen el mismo efecto en la pérdida de peso.

## ANOVA

**ANOVA** (*Analysis of Variance - Análisis de variación*) Es una técnica estadística utilizada para comparar las medidas de dos o más grupos. La idea detrás de ANOVA es descomponer la variabilidad total de los datos en dos componentes: variabilidad entre grupos y variabilidad dentro de los grupos:

- **Variabilidad entre grupos**: Esta variabilidad se refiere a las diferencias entre las medias del grupo. Si esta variabilidad es considerablemente mayor que la variabilidad dentro del grupo, podría ser una indicación de que al menos una de las medias del grupo es diferente.
- **Variabilidad dentro del grupo**: Esta variabilidad se refiere a la dispersión de los datos dentro de cada grupo. Si todos los grupos tienen una variabilidad similar, entonces cualquier diferencia notable en las medias de los grupos podría considerarse significativa.

Las hipótesis en ANOVA suelen incluir:

- **Hipótesis nula** ($H₀$): Las medias de todos los grupos son iguales.
- **Hipótesis alternativa** ($H₁$): Al menos una de las medias del grupo es diferente.

Si el resultado de la prueba ANOVA es significativo (por ejemplo, un valor p inferior a un umbral como 0,05), esto sugiere que al menos la media de un grupo es diferente.

## Ejercicio 2

Un agricultor decide probar tres tipos diferentes de fertilizantes para determinar si alguno es superior en términos de producción de maíz. El agricultor planta maíz en 15 parcelas idénticas y utiliza los tres fertilizantes (5 parcelas para cada tipo). Al final de la temporada mide el rendimiento de maíz (en kg) de cada parcela, con el siguiente resultado:

| Fertilizante 1 | Fertilizante 2 | Fertilizante 3 |
|:-------------|:-------------|:-------------|
| 20 | 22 | 24 |
| 21 | 21 | 23 |
| 20 | 23 | 22 |
| 19 | 22 | 23 |
| 20 | 21 | 24 |

Con estos datos busca responder la siguiente pregunta: ¿Existe una diferencia significativa en el rendimiento promedio del maíz entre los tres tipos de fertilizantes?

Para ayudarte, sigue los siguientes puntos:

-Enunciar la hipótesis: hipótesis nula y alternativa.
-Realizar la prueba ANOVA.
-Analizar las conclusiones.
-Si un fertilizante es mejor que otro, ¿cómo podemos saberlo?

In [42]:
# TODO

#H0 = No hay diferencia significativa en el rendimiento promedio del maíz entre los tres fertilizantes.
#H1 = Al menos un fertilizante tiene un rendimiento significativamente diferente.

fert_1 = [20, 21, 20, 19, 20]
fert_2 = [22, 21, 23, 22, 21]
fert_3 = [24, 23, 22, 23, 24]

f_stat, p_value = stats.f_oneway(fert_1, fert_2, fert_3)


print(f"t-value: {f_stat:.3f}")
print(f"p-value: {p_value:.3f}")

t-value: 20.316
p-value: 0.000


Los resultados obtenidos son:

Estadístico F = 20.32

p-valor = 0.00014

Usando un nivel de significancia común 

α=0.05:

Dado que el p-valor (0.00014) es menor que 0.05, rechazamos la hipótesis nula.

Esto indica que al menos un fertilizante tiene un rendimiento significativamente diferente a los demás.

In [46]:
import numpy as np
from statsmodels.stats.multicomp import pairwise_tukeyhsd

# Organizar datos para la prueba de Tukey
rendimiento = np.concatenate([fert_1, fert_2, fert_3])
grupo = (["Fertilizante 1"] * 5) + (["Fertilizante 2"] * 5) + (["Fertilizante 3"] * 5)

# Prueba de Tukey
tukey_result = pairwise_tukeyhsd(rendimiento, grupo, alpha=0.05)
tukey_result

# Mostrar los resultados de la prueba de Tukey
print(tukey_result)

       Multiple Comparison of Means - Tukey HSD, FWER=0.05        
    group1         group2     meandiff p-adj  lower  upper  reject
------------------------------------------------------------------
Fertilizante 1 Fertilizante 2      1.8 0.0099 0.4572 3.1428   True
Fertilizante 1 Fertilizante 3      3.2 0.0001 1.8572 4.5428   True
Fertilizante 2 Fertilizante 3      1.4 0.0409 0.0572 2.7428   True
------------------------------------------------------------------


Todos los fertilizantes presentan diferencias significativas entre sí.

Fertilizante 3 tiene el mayor rendimiento promedio y es significativamente mejor que los otros dos.

Fertilizante 1 tiene el menor rendimiento promedio.

Fertilizante 2 es intermedio, pero aún significativamente mejor que el Fertilizante 1 y peor que el Fertilizante 3.


🔹 El Fertilizante 3 es la mejor opción para maximizar la producción de maíz. 🌽